indico/modules/users/client/js/Favorites.jsx

```diff
@@ -0,0 +1,198 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import categoryFavoriteURL from 'indico-url:users.user_favorites_category_api';
+
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState, useCallback} from 'react';
+import ReactDOM from 'react-dom';
+import {Button, Icon, List, Loader, Popup} from 'semantic-ui-react';
+
+import {TooltipIfTruncated} from 'indico/react/components';
+import {PersonLinkSearch} from 'indico/react/components/principals/PersonLinkSearch';
+import {useFavoriteUsers} from 'indico/react/hooks';
+import {Translate} from 'indico/react/i18n';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+import './Favorites.module.scss';
+
+window.setupFavoriteSelection = function setupFavoriteSelection(userId) {
+  ReactDOM.render(<FavoriteManager userId={userId} />, document.getElementById('user-favorites'));
+};
+
+function FavoriteManager({userId}) {
+  const [favoriteUsers, [addFavoriteUser, deleteFavoriteUser], loading] = useFavoriteUsers(userId);
+
+  if (loading) {
+    return <Loader active />;
+  }
```

Comment:
I had it here because otherwise the category fav box jumps around after the user fav box is loaded. but I guess I can just place the loader further inside the divs to stop it from jumping around!